quarkus:info/update doesn't handle modules defined in activeByDefault profile

### Describe the bug

quarkus:info/update doesn't handle modules defined in activeByDefault profile

`mvn quarkus:info` works if `mvn clean install` is executed before the command (and thus the module defined in profile gets into local maven repository.

When running `mvn clean test` the module defined in activeByDefault profile gets picked, but running `mvn quarkus:info` command ends with `[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:2.7.1.Final:info (default-cli) on project module-c: Failed to resolve the Quarkus application model for project org.acme:module-c::pom:1.0.0-SNAPSHOT: Failed to resolve artifact org.acme:module-c:pom:1.0.0-SNAPSHOT: Could not find artifact org.acme:module-c:pom:1.0.0-SNAPSHOT -> [Help 1]`

part of the stacktrace when using `mvn -e ...`
```
Caused by: org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.acme:module-c:pom:1.0.0-SNAPSHOT
    at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve (DefaultArtifactResolver.java:414)
    at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts (DefaultArtifactResolver.java:229)
    at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifact (DefaultArtifactResolver.java:207)
    at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveArtifact (DefaultRepositorySystem.java:262)
    at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.resolveInternal (MavenArtifactResolver.java:164)
    at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.resolve (MavenArtifactResolver.java:158)
    at io.quarkus.bootstrap.resolver.BootstrapAppModelResolver.resolve (BootstrapAppModelResolver.java:323)
    at io.quarkus.bootstrap.resolver.BootstrapAppModelResolver.doResolveModel (BootstrapAppModelResolver.java:195)
    at io.quarkus.bootstrap.resolver.BootstrapAppModelResolver.resolveManagedModel (BootstrapAppModelResolver.java:166)
    at io.quarkus.bootstrap.resolver.BootstrapAppModelResolver.resolveModel (BootstrapAppModelResolver.java:149)
    at io.quarkus.maven.QuarkusProjectStateMojoBase.resolveApplicationModel (QuarkusProjectStateMojoBase.java:69)
    at io.quarkus.maven.InfoMojo.processProjectState (InfoMojo.java:33)
    at io.quarkus.maven.QuarkusProjectStateMojoBase.doExecute (QuarkusProjectStateMojoBase.java:56)
    at io.quarkus.maven.QuarkusProjectMojoBase.execute (QuarkusProjectMojoBase.java:111)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
```

@aloubyansky 

### Expected behavior

quarkus:info/update executed properly on modules defined in activeByDefault profile

### Actual behavior

quarkus:info/update is not executed properly on modules defined in activeByDefault profile, ends with failure to resolve artifact

### How to Reproduce?

 - clone https://github.com/rsvoboda/code-with-quarkus-multi (module-c is defined in profile that is activatedByDefault)
 - run `mvn quarkus:info`


### Output of `uname -a` or `ver`

macOS

### Output of `java -version`

Java 17

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.7.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.1

### Additional information

_No response_

--- a/devtools/maven/src/main/java/io/quarkus/maven/QuarkusProjectStateMojoBase.java
+++ b/devtools/maven/src/main/java/io/quarkus/maven/QuarkusProjectStateMojoBase.java
@@ -50,11 +50,20 @@ public abstract class QuarkusProjectStateMojoBase extends QuarkusProjectMojoBase
             throw new MojoExecutionException("This goal requires a Quarkus extension registry client to be enabled");
         }
 
+        // to support profiles
+        final String emb = System.getProperty("quarkus.bootstrap.effective-model-builder");
+        System.setProperty("quarkus.bootstrap.effective-model-builder", "true");
+
         final Collection<Path> createdDirs = ensureResolvable(new DefaultArtifact(project.getGroupId(),
                 project.getArtifactId(), ArtifactCoords.TYPE_POM, project.getVersion()));
         try {
             processProjectState(quarkusProject);
         } finally {
+            if (emb == null) {
+                System.clearProperty("quarkus.bootstrap.effective-model-builder");
+            } else {
+                System.setProperty("quarkus.bootstrap.effective-model-builder", emb);
+            }
             for (Path p : createdDirs) {
                 IoUtils.recursiveDelete(p);
             }

